project/npda/general_functions/organisations_adapter.py

```diff
@@ -82,3 +88,105 @@ def organisations_to_populate_select_field(request, user_instance=None):
                 (org.ods_code, org.name) for org in sibling_organisations
             ]
     return organisation_choices
+
+
+def paediatric_diabetes_units_to_populate_select_field(request, user_instance=None):
+    """
+    This function is used to populate the add_employer field with paediatric diabetes units that the user_instance is NOT already affiliated with, based on request user permissions.
+    If no user_instance is provided (as the user form is being created), the function will populate the add_employer field with all paediatric diabetes units that the request user has access to.
+    """
+
+    PaediatricDiabetesUnit = apps.get_model("npda", "PaediatricDiabetesUnit")
+
+    if user_instance:
+        if (
+            request.user.is_superuser
+            or request.user.is_rcpch_audit_team_member
+            or request.user.is_rcpch_staff
+        ):
+            # return all paediatric diabetes units excluding those were the user is employed
+            return (
+                PaediatricDiabetesUnit.objects.all()
+                .exclude(npdauser=request.user)
+                .order_by("organisation_name")
+                .annotate(
+                    paediatric_diabetes_unit_name=Concat(
```

Comment:
I'm not normally too hung up on DRY code but I think perhaps a function to build each of the annotate arguments would be useful to highlight what the differences are between each one?

It's hard to eyeball each one individually and understand what it's doing differently to the others

Reply:
fair point. sorry sent this in possibly before it was finished. Not every select needs the same list of PDUs - that is what this trying to do. I will fix and let you know when ready to review.